pub.go

```diff
@@ -213,15 +240,46 @@ type pubMWriter struct {
 	ctx context.Context
 	mu  sync.Mutex
 	ws  []*Conn
+
+	qmu    sync.Mutex
+	qcond  *sync.Cond
+	q      *Queue
+	hwm    int
+	closed bool
 }
 
 func newPubMWriter(ctx context.Context) *pubMWriter {
-	return &pubMWriter{
+	p := &pubMWriter{
 		ctx: ctx,
+		q:   NewQueue(),
+	}
+	p.qcond = sync.NewCond(&p.qmu)
+	go p.run()
```

Comment:
it's not completely clear to me when would/should this goroutine shutdown.
shouldn't we somehow couple it with the "lifeline" ctx?

Reply:
The goroutine is shutdown when the pub socket is closed via `pubSocket.Close`. This can also occur when the context is no longer valid as the user will (or at least should) close the socket then.